packages/utils/src/composeAbortSignals.ts

```diff
@@ -10,29 +10,42 @@ export type ComposedAbortSignal = AbortSignal & { destroy: () => void }
  * aforementioned instance of AbortSignal will have more and more listeners added
  * but never cleaned.
  */
-export function composeAbortSignals(...signals: AbortSignal[]): ComposedAbortSignal {
-    if (signals.length === 0) {
-        throw new Error('must provide at least one AbortSignal')
-    }
+export function composeAbortSignals(
+    ...signals: (AbortSignal | undefined)[]
+): ComposedAbortSignal {
+    const abortController = new AbortController()
+
+    for (const signal of signals) {
+        if (signal?.aborted) {
+            abortController.abort()
```

Comment:
The above `abortController` is in the right place. After all, its signal is being used in both "returns".